### DISCUSSION TITLE
Estimated Reading Time

### DISCUSSION THREAD

**Discussion by trueberryless** [withastro/starlight#1267] [Feature Requests]
### What version of `starlight` are you using?

0.15.0

### What is your idea?

Estimate the time the user needs to read the whole page. Calculate this by counting the words on the page assuming around 200 words can be read per minute.

I discovered, that there is now a [plugin feature](#962) in Starlight, so maybe this can be implemented with such plugins...

### Why is this feature necessary?

Nice to have, not necessary. 

### Do you have examples of this feature in other projects?

- medium.com has something like that I think...
![image](https://github.com/withastro/starlight/assets/99918022/7b0e91c1-14c1-4980-9b8c-596aa2a6690b)

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by HiDeoo** [comment]
> I discovered, that there is now a [plugin feature](https://github.com/withastro/starlight/discussions/1267#962) in Starlight, so maybe this can be implemented with such plugins...

Thanks for the feedback 🙌 

Considering a reading time is fairly less common on documentation vs blog posts for example, I think it would indeed be a good idea to have such a feature implemented as a plugin, so that it can be used by those who need it only. Imagining such a plugin, I think all the tools are already there in Starlight to implement it.
